test/script/RebalanceScUsdcV2.s.t.sol

```diff
@@ -0,0 +1,532 @@
+
+        script.run();
+
+        assertApproxEqRel(vault.usdcBalance(), expectedFloat, 0.001e18, "usdc balance");
+        assertApproxEqRel(vault.wethInvested(), expectedDebt, 0.001e18, "weth invested");
+        assertApproxEqRel(vault.totalDebt(), expectedDebt, 0.001e18, "total debt");
+        assertApproxEqRel(vault.totalCollateral(), expectedCollateral, 0.001e18, "total collateral");
+    }
+
+    function test_run_leverageDownByAddingMoreCollateralAndRepayingDebt() public {
+        script.run(); // initial rebalance
+        script = new RebalanceScUsdcV2TestHarness(); // reset script state
+
+        assertTrue(vault.wethInvested() > 0, "weth invested");
+        assertTrue(vault.totalDebt() > 0, "total debt");
+        assertTrue(vault.totalCollateral() > 0, "total collateral");
+        assertTrue(vault.usdcBalance() > 0, "usdc balance");
+
+        // additional deposit
+        deal(address(vault.asset()), address(this), 1e6);
+        vault.asset().approve(address(vault), 1e6);
+        vault.deposit(1e6, address(this));
+
+        script.setMorphoTargetLtv(script.morphoTargetLtv() - 0.5e18);
+
+        uint256 expectedFloat = vault.totalAssets().mulWadDown(vault.floatPercentage());
+        uint256 expectedCollateral = vault.totalAssets() - expectedFloat;
+        uint256 expectedDebt = priceConverter.usdcToEth(expectedCollateral).mulWadDown(script.morphoTargetLtv());
+
+        script.run();
+
+        assertApproxEqRel(vault.wethInvested(), expectedDebt, 0.001e18, "weth invested");
+        assertApproxEqRel(vault.totalDebt(), expectedDebt, 0.001e18, "total debt");
+        assertApproxEqRel(vault.totalCollateral(), expectedCollateral, 0.001e18, "total collateral");
+    }
+
+    function test_run_leverageUpByTakingMoreDebt() public {
+        script.setMorphoTargetLtv(0.5e18);
+        script.run(); // initial rebalance
+        script = new RebalanceScUsdcV2TestHarness(); // reset script state
+
+        assertTrue(vault.wethInvested() > 0, "weth invested");
+        assertTrue(vault.totalDebt() > 0, "total debt");
+        assertTrue(vault.totalCollateral() > 0, "total collateral");
+        assertTrue(vault.usdcBalance() > 0, "usdc balance");
+
+        script.setMorphoTargetLtv(0.6e18);
+
+        uint256 expectedFloat = vault.totalAssets().mulWadDown(vault.floatPercentage());
+        uint256 expectedCollateral = vault.totalAssets() - expectedFloat;
+        uint256 expectedDebt = priceConverter.usdcToEth(expectedCollateral).mulWadDown(script.morphoTargetLtv());
+
+        script.run();
+
+        assertApproxEqRel(vault.wethInvested(), expectedDebt, 0.001e18, "weth invested");
+        assertApproxEqRel(vault.totalDebt(), expectedDebt, 0.001e18, "total debt");
+        assertApproxEqRel(vault.totalCollateral(), expectedCollateral, 0.001e18, "total collateral");
+    }
+
+    function test_run_restoresMissingFloat() public {
```

Comment:
There should be an assertion for the missing float check here right?

Reply:
[added](https://github.com/lindy-labs/sandclock-contracts/pull/118/commits/0613abc0fdbf60f8fbeb482318dfe57129f8511a)